### ISSUE TITLE
Normalize resource

### ISSUE THREAD

**Issue by akariv** [datahq/dataflows#84]
Add ability to normalize a resource - i.e. extract a few columns out of a resource into a 'lookup table' resource, add proper indexes and pointers and create foreign key relationships between the two.

**Comment by rufuspollock** [comment]
@akariv can you give a bit more detail - i normally think of "normalize" as like "unpivot" cf #72

**Comment by akariv** [comment]
Yes - unpivoting is actually a process for *denormalizing* data.

Normalizing is the process of removing duplication, as can be read about here:
https://en.wikipedia.org/wiki/Database_normalization

**Comment by rufuspollock** [comment]
Got it and clear. Thanks :smile:
